modules/SofaGeneralLoader/SphereLoader.cpp

```diff
@@ -39,19 +41,62 @@ int SphereLoaderClass = core::RegisterObject("Loader for sphere model descriptio
 
 SphereLoader::SphereLoader()
     :BaseLoader(),
-     positions(initData(&positions,"position","Sphere centers")),
-     radius(initData(&radius,"listRadius","Radius of each sphere")),
-     d_scale(initData(&d_scale,"scale","Scale applied to sphere positions & radius")),
-     d_translation(initData(&d_translation,"translation","Translation applied to sphere positions"))
+     d_positions(initData(&d_positions,"position","Sphere centers")),
+     d_radius(initData(&d_radius,"listRadius","Radius of each sphere")),
+     d_scale(initData(&d_scale, Vec3(1.0, 1.0, 1.0), "scale","Scale applied to sphere positions & radius")),
+     d_rotation(initData(&d_rotation, Vec3(), "rotation", "Rotation of the DOFs")),
+     d_translation(initData(&d_translation, Vec3(),"translation","Translation applied to sphere positions"))
+{
+    addAlias(&d_positions,"sphere_centers");
+    addAlias(&d_scale, "scale3d");
 
+    addUpdateCallback("filename", { &m_filename }, [this](const core::DataTracker& )
+    {
+        if (load()) {
+            clearLoggedMessages();
+            return sofa::core::objectmodel::ComponentState::Valid;
+        }
+        return sofa::core::objectmodel::ComponentState::Invalid;
+    }, { &d_positions, &d_radius });
+
+    addUpdateCallback("updateTransformPosition", { &d_translation, &d_rotation, &d_scale}, [this](const core::DataTracker&)
+    {
+        applyTransform();
+        return sofa::core::objectmodel::ComponentState::Valid;
+    }, { &d_positions });
+}
+
+
+void SphereLoader::applyTransform()
 {
-    addAlias(&positions,"sphere_centers");
+    const Vec3& scale = d_scale.getValue();
+    const Vec3& rotation = d_rotation.getValue();
+    const Vec3& translation = d_translation.getValue();
+    m_internalEngine["updateTransformPosition"].cleanDirty();
```

Comment:
Adding DataCallBack clean inside the method avoid the callback to be relaunched later. And give the good call stack:
```
SphereLoader::load
SphereLoader::applyTransform
SphereLoader::applyTransform exit
SphereLoader::load exit
```

Any suggestion @hugtalbot @marques-bruno to do that in a better way.